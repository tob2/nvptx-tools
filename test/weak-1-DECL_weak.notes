Given "error: PTX does not support weak declarations (only weak definitions)",
GCC/nvptx never emits a "weak" DECL alone.  It doesn, however, emit a "weak"
GLOBAL FUNCTION DECL before a "strong" GLOBAL FUNCTION DECL, so we have to
handle "weak" DECLs anyway, and therefore also want to test it individually.
