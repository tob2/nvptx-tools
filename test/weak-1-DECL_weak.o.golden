// BEGIN PREAMBLE
.version 4.0
.target sm_50
.address_size 64
// END PREAMBLE
// BEGIN GLOBAL FUNCTION DECL: __gbl_ctors
.weak .func __gbl_ctors;
// BEGIN GLOBAL VAR DECL: __exitval_ptr
.weak .global .align 8 .u64 __exitval_ptr[1];
